src/mongo/client/dbclient.cpp

```diff
@@ -1066,15 +1070,175 @@ namespace mongo {
         return runCommand(db.c_str(), b.done(), *info);
     }
 
-    bool DBClientWithCommands::copyDatabase(const string &fromdb, const string &todb, const string &fromhost, BSONObj *info) {
+    string DBClientWithCommands::_authKeyCopyDB(const string &username,
```

Comment:
Can we make the name more descriptive. At least indicate that it's MONGODB-CR specific.